### DISCUSSION TITLE
Private (unexported) method

### DISCUSSION THREAD

**Discussion by KilianVounckx** [vlang/v#9498] [Ideas]
As I see it right now (I could be wrong), a function is private by default and is has to be made public with the pub keyword. Methods however seem to be public. Is there a reason for this? And is there a way to make them private?
I would think that having them behave like functions is the most logical.

```v
fn (t MyType) foo() {}
```

The method above  can be called by any other module. However I think it would be more  logical to have that method  be private and have the developer explicitly make them public, like functions:

```v
pub fn (t MyType) foo() {}
```

**Comment by medvednikov** [comment]
Both functions and methods are private by default.

**Reply by KilianVounckx** [reply]
In that case I will double check and try again later today, but I thought I was able to use the methods outside the module.

**Reply by medvednikov** [reply]
I just tested, and both fns and methods work correctly.

```
/tmp $ v x.v
x.v:3:3: error: method `os.File.foo` is private
    1 | import os
    2 | f := os.open('f')?
    3 | f.foo()
      |   ~~~~~
```

**Reply by KilianVounckx** [reply]
For some reason this is not the case with my simple program. I made it as simple as possible by only having one type and one 'private' method.
I have a folder like this:

rtc
|__tuple.v
|__tuple_test.v

The contents of tuple.v are now:
```v
module rtc

type Tuple = []f64

pub fn new_tuple(x f64, y f64, z f64, w f64) Tuple {
    return Tuple([x, y, z, w])
}

fn (t Tuple) x() f64 {
    return t[0]
}
```

tuple_test.v is ever simplere:
```v
module rtc_test

import rtc {new_tuple}

fn test_new_tuple() {
    a := new_tuple(4.3, -4.2, 3.1, 1)
    assert a.x() == 4.3
}
```

when i run (from within the rtc directory);
```shell
v test tuple_test.v
```
the program compiles, and the test passes.

I was expecting the compiler to halt because the rtc_test modul should not be able to have access to the x method.
Maybe it has something to do with my file structure?

**Reply by medvednikov** [reply]
This is a bug, thanks for reporting.

**Reply by medvednikov** [reply]
Looks like it only applies to local modules.
